src/hooks/useSafeTokenAllocation.ts

```diff
@@ -66,10 +78,7 @@ const completeAllocation = async (allocation: VestingData): Promise<Vesting> =>
   }
 
   // Allocation is not yet redeemed => check the redeemDeadline
-  const redeemDeadline = await web3ReadOnly.call({
-    to: allocation.contract,
-    data: airdropInterface.encodeFunctionData('redeemDeadline'),
-  })
+  const redeemDeadline = await _getRedeemDeadline(allocation, web3ReadOnly)
```

Comment:
We could also cache `isRedeemed` if it is true as it cannot change once the user redeemed to improve it a bit further.

Reply:
Nevermind. We need to fetch again as the `amountClaimed` could be changed!